meteor/server/api/rest/api.ts

```diff
@@ -0,0 +1,1494 @@
+				event,
+				getCurrentTime(),
+				rundownPlaylistId,
+				() => {
+					check(rundownPlaylistId, String)
+				},
+				StudioJobs.RemovePlaylist,
+				{
+					playlistId: rundownPlaylistId,
+				}
+			)
+		)
+
+		await Promise.all(promises)
+		Studios.remove(studioId)
+
+		return ClientAPI.responseSuccess(undefined, 200)
+	}
+}
+
+const koaRouter = new KoaRouter()
+
+function extractErrorCode(e: unknown): number {
+	if (ClientAPI.isClientResponseError(e)) {
+		return e.errorCode
+	} else if (UserError.isUserError(e)) {
+		return e.errorCode
+	} else if ((e as Meteor.Error).error && typeof (e as Meteor.Error).error === 'number') {
+		return (e as Meteor.Error).error as number
+	} else {
+		return 500
+	}
+}
+
+function extractErrorMessage(e: unknown): string {
+	if (ClientAPI.isClientResponseError(e)) {
+		return translateMessage(e.error.message, interpollateTranslation)
+	} else if (UserError.isUserError(e)) {
+		return translateMessage(e.message, interpollateTranslation)
+	} else if ((e as Meteor.Error).reason && typeof (e as Meteor.Error).reason === 'string') {
+		return (e as Meteor.Error).reason as string
+	} else {
+		return (e as Error).message ?? 'Internal Server Error' // Fallback in case e is not an error type
+	}
+}
+
+function sofieAPIRequest<Params, Body, Response>(
+	method: 'get' | 'post' | 'put' | 'delete',
+	route: string,
+	errMsgs: Map<number, UserErrorMessage>,
+	handler: (
+		serverAPI: RestAPI,
+		connection: Meteor.Connection,
+		event: string,
+		params: Params,
+		body: Body
+	) => Promise<ClientAPI.ClientResponse<Response>>
+) {
+	koaRouter[method](route, async (ctx, next) => {
+		try {
+			const serverAPI = new ServerRestAPI()
+			const response = await handler(
+				serverAPI,
+				makeConnection(ctx),
+				restAPIUserEvent(ctx),
+				ctx.params as unknown as Params,
+				ctx.request.body as unknown as Body
+			)
+			if (ClientAPI.isClientResponseError(response)) throw response
+			ctx.body = JSON.stringify({ status: response.success, result: response.result })
+			ctx.status = response.success
+		} catch (e) {
+			const errCode = extractErrorCode(e)
+			const errMsg = extractErrorMessage(UserError.create(errMsgs.get(errCode) || UserErrorMessage.InternalError, undefined, errCode))
+
+			logger.error(`${method.toUpperCase()} failed for route ${route}: ${errCode} - ${errMsg}`)
+			ctx.type = 'application/json'
+			ctx.body = JSON.stringify({ status: errCode, message: errMsg })
+			ctx.status = errCode
+		}
+		await next()
+	})
+}
+
+koaRouter.get('/', async (ctx, next) => {
+	ctx.type = 'application/json'
+	const server = new ServerRestAPI()
+	const response = ClientAPI.responseSuccess(await server.index())
+	ctx.body = JSON.stringify({ status: response.success, result: response.result })
+	ctx.status = response.success
+	await next()
+})
+
+sofieAPIRequest<never, never, string[]>('get', '/playlists', new Map(), async (serverAPI, connection, event, _params, _body) => {
+	logger.info(`koa GET: playlists`)
```

Comment:
I guess this is petty, but I don't think this should say `koa`, since that's an implementation detail that will be likely unfamiliar to someone reading these logs.

Reply:
I've changed the logs to log e.g. `API GET: ...` so we can search logs for `API [METHOD]`